extensions/sql/contract-definition/loader/src/test/java/org/eclipse/dataspaceconnector/sql/contractdefinition/loader/SqlContractDefinitionLoaderTest.java

```diff
@@ -0,0 +1,138 @@
+/*
+ *  Copyright (c) 2022 Daimler TSS GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Daimler TSS GmbH - Initial Tests
+ *
+ */
+
+package org.eclipse.dataspaceconnector.sql.contractdefinition.loader;
+
+import org.eclipse.dataspaceconnector.dataloading.ContractDefinitionLoader;
+import org.eclipse.dataspaceconnector.junit.launcher.EdcExtension;
+import org.eclipse.dataspaceconnector.policy.model.Policy;
+import org.eclipse.dataspaceconnector.spi.asset.AssetSelectorExpression;
+import org.eclipse.dataspaceconnector.spi.system.ConfigurationExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.system.configuration.ConfigFactory;
+import org.eclipse.dataspaceconnector.spi.transaction.TransactionContext;
+import org.eclipse.dataspaceconnector.spi.transaction.datasource.DataSourceRegistry;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+import org.eclipse.dataspaceconnector.sql.SqlQueryExecutor;
+import org.eclipse.dataspaceconnector.sql.contractdefinition.spi.ConfigurationKeys;
+import org.eclipse.dataspaceconnector.sql.contractdefinition.spi.SqlContractDefinitionTables;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.dataspaceconnector.sql.SqlQueryExecutor.executeQuery;
+
+@ExtendWith(EdcExtension.class)
+class SqlContractDefinitionLoaderTest {
```

Comment:
Why wouldn't you just instantiate the `SqlContractDefinitionLoader` with explicitly instantiated parameters for `DataSource`, `TransactionContext` etc? Testing the loader against H2 should not require instantiating the `ConfigurationExtension` and spinning up the entire DI container plus registering a local extension to initialize the database.
This test has to be rewritten IMO.